src/lex/mod.rs

```diff
@@ -923,3 +600,365 @@ impl Iterator for Lexer {
         c.or_else(|| self.error_handler.pop_front().map(Err))
     }
 }
+
+pub(crate) trait LiteralParser {
+    fn next_char(&mut self) -> Option<char>;
+    fn peek(&mut self) -> Option<char>;
+    fn peek_next(&mut self) -> Option<char>;
+    fn get_location(&self) -> &SingleLocation;
+    fn handle_error(&mut self, err: Locatable<LexError>);
+    fn handle_warning(&mut self, err: Locatable<Warning>);
+
+    fn handle_error_loc<E: Into<LexError>>(&mut self, err: E, location: Location) {
+        self.handle_error(location.with(err.into()));
+    }
+    fn handle_warning_loc<W: Into<Warning>>(&mut self, warning: W, location: Location) {
+        self.handle_warning(location.with(warning.into()));
+    }
+
+    /// Given the start of a span as an offset,
+    /// return a span lasting until the current location in the file.
+    fn span(&self, start: u32) -> Location {
+        Location {
+            span: (start..self.get_location().offset).into(),
+            file: self.get_location().file,
+        }
+    }
+    /// If the next character is `item`, consume it and return true.
+    /// Otherwise, return false.
+    fn match_next(&mut self, item: char) -> bool {
+        if self.peek().map_or(false, |c| c == item) {
+            self.next_char();
+            true
+        } else {
+            false
+        }
+    }
+    /// Read a logical character, which may be a character escape.
+    ///
+    /// Has a side effect: will call `warn` if it sees an invalid escape.
+    ///
+    /// Before: u8s{"\b'"}
+    /// After:  chars{"'"}
```

Comment:
Whoops, this is left over from a bad change in https://github.com/jyn514/saltwater/pull/468 
```suggestion
    /// Before: chars{"\b'"}
    /// After:  chars{"'"}
```